### ISSUE TITLE
Problems With php short_tags

### ISSUE THREAD

**Issue by BrunoMarinhoM** [phpactor/phpactor#2697]
Hello maintainers! 

First of all, thank you all for the amazing work on this LS and on every tool phpactor offers! It really does help me and many people a lot! 

---
- On neovim 0.10 (but the behavior also showed in my older version);
- nvim/lsp-config plug in;

# The Behavior: 

The problem I'm facing is after seting  phpactor's LSP only parses propertly when the .php file starts with the full tag (<?php);

Otherwise, it will not give any information about any php within that file;

Also, if the php does not start in with the full tag and other file import/use it, it won't be tracked.

![Captura de tela de 2024-07-17 12-59-04](https://github.com/user-attachments/assets/b4c7b6a7-3364-4162-9695-6ed911d606b9)

![Captura de tela de 2024-07-17 12-58-52](https://github.com/user-attachments/assets/45ee81ec-0ba8-4d63-8970-1a9ed6be9c8c)

As shown in the images above, the only info phpactor provides is InlineHTML.

I'm initializing phpactor with the default configs.

---

**Comment by dantleech** [comment]
Hmm, I'd that valid PHP?https://www.php.net/manual/en/language.basic-syntax.phptags.php

**Comment by BrunoMarinhoM** [comment]
Thank you for the reply! 

I made the image just as an example! To show the behavior.

![Captura de tela de 2024-07-17 15-55-04](https://github.com/user-attachments/assets/d206d038-6037-499f-8c85-15095157a13c)

![Captura de tela de 2024-07-17 15-54-55](https://github.com/user-attachments/assets/8b37c669-b4ca-48c4-9215-cec9bf1e3538)

![image](https://github.com/user-attachments/assets/ba60c0f8-f990-4636-bf6c-8ce09bb9776b)
![Captura de tela de 2024-07-17 16-00-52](https://github.com/user-attachments/assets/1f981bf6-038a-4078-af98-ed504426882c)


Those 4 images i think may be better to show the behavior and are actually copies from the https://www.php.net/manual/en/language.basic-syntax.phptags.php; 

I've been coding in php only for a couple of months now so maybe my examples were not as good, but in the code-base I'm working on there's plenty of php that is all running propertly (and is valid I assume) but phpactor does not seem to recognize as valid php;

Surely it may not be a bug but just a configuration I missed! But I could not find any resource on it! If you could help me I'd be very grateful.

Thank you! :)

**Comment by dantleech** [comment]
hmm works for me:

![image](https://github.com/user-attachments/assets/58f27087-72bc-4b84-adf6-5724e6534a6f)

**Comment by dantleech** [comment]
can you paste a "broken" code example here in a [fenced code block[(https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) or in a gist?

**Comment by BrunoMarinhoM** [comment]
Sure!

Just for context;

I'm using phpactor version 2024.06.30.0;

![Captura de tela de 2024-07-17 19-31-05](https://github.com/user-attachments/assets/82c19c14-e3d9-4262-8ac0-afd8eb96db2b)

I've installed it through lsp-config 

`LspInstall phpactor` nvim command;

and, in my init.lua config file I've the following 

```
require('lspconfig').phpactor.setup {
  on_attach = on_attach,
  init_options = {
    ['language_server_phpstan.enabled'] = false,
    ['language_server_psalm.enabled'] = false,
  },
}
```

[as recomended](https://phpactor.readthedocs.io/en/master/lsp/vim.html?highlight=nvim) for vim;

(The mason/lsp-config configurations are the same as the [kick-start.nvim](https://github.com/nvim-lua/kickstart.nvim))
(Also, I've tried other configs and None but it keeps not working with short tags);


As for the examples:

### Works only within the <?php ?> tags and not the ones bellow
```
<?php 

$i = 0;
echo 'if you want to serve PHP code in XHTML or XML documents,';

?>


<? 

$i = 0;
echo 'if you want to serve PHP code in XHTML or XML documents,';

?>
```


### From my code base, also same problem
```
<?
namespace FaturaDao;
{

    public $con;
    function __construct($con)
    {
        $this->con = $con;
    }
}

```

It is any code that uses php's short_tag, so i could generate more of them, though i do not think it would be insightful;

**Comment by dantleech** [comment]
I think I can reproduce this:

```
php -dshort_open_tag=Off ./bin/phpactor worse:dump-ast test.php
```

Note that if you change `short_open_tag=On` then you can see the nodes being parsed in the short tag.

I think you need to edit your `php.ini` on your system and set `short_open_tag=On`

Note: I'm really not sure why the parser uses ini settings like this, but c'est la vie.

**Comment by BrunoMarinhoM** [comment]
Yes. Indeed I had to change the global php.ini file to allow it to work with the short_open_tag's

I tried to change only within the `/home/USER/.local/share/nvim/opt/phpactor/docker/php.ini`
and/or `/home/USER/.local/share/nvim/lazy/phpactor/docker/php.ini` but none took effect!

As I was trying to make it work I also realized that there's a error when parsing files like:

```
<?
namespace ClassNameSpace;
(empty line)
use OtherClass\OtherClass;
(empty line)
class ClassExample
```

with empty lines between those initial declarations. The error was specifically when importing files like those;

But If i do not find any solution, I'll open another issue to discuss.

Thank you!
